### PR TITLE
Feature/post backend

### PR DESCRIPTION
**Pull request by danielcfuentes**
**Description**
This pull request is setting up the backend for a user to create posts as well as be able to retrieve all posts made by all users.
I created the schema for my tables for the posts and images. I then created the endpoints for creating a post and retireving all the posts. I integrate the user's access token as the middleware to be able to do these requests

what will be done in later PRs and not included here:
Integrate the backend requests with the frontend for a user to be able to fill out the modal for creating a post and the post to be created. Then be able to see al the posts on the feed page. 

**Milestones**
The milestones or stories/features that this works towards:
Milestone 2: Creating Posts for Running Route and mages

**Test Plan**
creating a post with user "luis" logged in:
image

creating a second post :
image

prisma dashboard:
image

retrieving all posts as user "bryan" logged in:
image

### CODE REVIEW

File: backend/prisma/migrations/20240702184800_posts_image_tables/migration.sql [hunk 1]
```diff
@@ -0,0 +1,26 @@
+-- CreateTable
+CREATE TABLE "Post" (
+    "id" SERIAL NOT NULL,
+    "title" TEXT NOT NULL,
+    "content" TEXT NOT NULL,
```
Comment (jayeshdoolani) [inline backend/prisma/migrations/20240702184800_posts_image_tables/migration.sql:5]: _nit_: best to add some character limits

File: backend/prisma/migrations/20240702184800_posts_image_tables/migration.sql [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    "createdAt" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    "updatedAt" TIMESTAMP(3) NOT NULL,
```
Comment (jayeshdoolani) [inline backend/prisma/migrations/20240702184800_posts_image_tables/migration.sql:7]: when a post is created, is it also the same time when it's updated? If not, then shouldn't this be nullable?

File: backend/server.js [hunk 3]
```diff
@@ -22,4 +21,57 @@ function authenticateToken(req, res, next) {
...
+    res.json(posts);
+  } catch (error) {
+    res
+      .status(500)
+      .json({ message: "Error fetching all posts", error: error.message });
+  }
+});
+
+app.listen(PORT, () => {
+  console.log(`Server is running on http://localhost:${PORT}`);
```
Comment (jayeshdoolani) [inline backend/server.js]: No need to mention which port the website is running on; Security POV

File: backend/server.js [hunk 4]
```diff
@@ -2,14 +2,13 @@ require("dotenv").config();
 const express = require("express");
 const app = express();
 const jwt = require("jsonwebtoken");
-app.use(express.json);
+const { PrismaClient } = require("@prisma/client");
+const prisma = new PrismaClient();
+const PORT = 3000;
```
Comment (jayeshdoolani) [inline backend/server.js]: does this need to be in code?

**Review by jayeshdoolani — APPROVED** [review]

File: backend/server.js [hunk 5]
```diff
@@ -22,4 +21,57 @@ function authenticateToken(req, res, next) {
...
+  try {
+    const newPost = await prisma.post.create({
+      data: {
+        title,
+        content,
+        user: { connect: { username: req.user.name } },
+        images: {
+          create: imageUrls.map((url) => ({ url })),
+        },
+      },
```
Comment (gwh59) [inline backend/server.js]: Does `prisma` handle data sanitization for you? Might be outside the scope of this PR but just wanted to make sure that's something that's being handled

**Review by gwh59 — APPROVED** [review]

File: backend/server.js (hunk 5 above)
Comment (danielcfuentes) [reply backend/server.js]: I looked it up to check and this what i found:
https://github.com/prisma/prisma/discussions/7635

File: backend/server.js (hunk 3 above)
Comment (danielcfuentes) [reply backend/server.js]: Just went back and removed it from both files and will keep this in mind.

File: backend/prisma/migrations/20240702184800_posts_image_tables/migration.sql (hunk 2 above)
Comment (danielcfuentes) [reply backend/prisma/migrations/20240702184800_posts_image_tables/migration.sql:7]: Correct, later on i wanted to implement for a user to be able to update and edit posts.

Comment (danielcfuentes) [reply backend/prisma/migrations/20240702184800_posts_image_tables/migration.sql:7]: updatedAt is now defined as DateTime?, indicating that it is optional.
These changes ensure that the updatedAt field is nullable at creation but will be updated with the current timestamp whenever the post is updated.

File: backend/prisma/migrations/20240702184800_posts_image_tables/migration.sql (hunk 1 above)
Comment (danielcfuentes) [reply backend/prisma/migrations/20240702184800_posts_image_tables/migration.sql:5]: added character limits using @db.VarChar()

**Review by janicetsai1 — APPROVED** [review]
thanks for addressing the updates

File: backend/server.js (hunk 4 above)
Comment (danielcfuentes) [reply backend/server.js]: Moved it to the env file and then import it into my server files:
const PORT = process.env.SERVER_PORT;

**Pull request merged**